src/scenes/ProgressReports/EditForm/Steps/ExplanationOfProgress/ExplanationOfProgress.tsx

```diff
@@ -170,8 +170,10 @@ export const ExplanationOfProgress: StepComponent = ({ report }) => {
               <EnumField
                 name="reasons"
                 label="Select a reason"
-                options={optionsByGroup[group]}
                 required
+                options={optionsByGroup[group].filter(
+                  (reason) => !optionsByGroup.deprecated.includes(reason)
+                )}
```

Comment:
This is functionally the same as what we had in a previous iteration where the API just removed this value from the list. We want to show this value, just as disabled. Here's what I would do:
```ts
<EnumField
  name="reasons"
  label="Select a reason"
  required
  layout="column"
  disabled={!explanation.reasons.canEdit}
>
  {optionsByGroup[group].flatMap((reason) => {
    const isDeprecated =
      optionsByGroup.deprecated.includes(reason);
    if (isDeprecated && reason !== reasons) {
      // Don't even show deprecated options if they are not currently selected
      return [];
    }
    return (
      <EnumOption
        key={reason}
        label={reason}
        value={reason}
        disabled={isDeprecated}
      />
    );
  })}
</EnumField>
```

Reply:
You've lost me on the if check `isDeprecated && reason !== reasons` where does reasons come from?

Reply:
Also, why would we return an empty array in that check when the map return of that flatMap returns an Element?

Reply:
Oh my b. `reasons` is destructured from `values` on L126.
flatMap can return one or more elements which are flattened into a single list. So returning `[]` basically means skip this input item.